src/main/java/duke/task/Task.java

```diff
@@ -0,0 +1,73 @@
+package duke.task;
+
+import java.time.LocalDate;
+
+/**
+ * Class for Task object to encapsulate commonalities in subclasses.
+ *
+ * @author Andre Lin HuiKai
+ * @version CS2103T AY22/23 Semester 2
+ */
+public abstract class Task {
+    private String description;
+    private boolean isDone;
+
+    /**
+     * Constructor for Task object.
+     * @param des Description of the task.
+     *
+     * @author Andre Lin HuiKai
+     * @version CS2103T AY22/23 Semester 2
+     */
+    public Task(String des) {
+        this.description = des;
+        this.isDone = false;
+    }
+
+    /**
+     * Marks the task as completed.
+     */
+    public void markTask() {
+        this.isDone = true;
+    }
+
+    /**
+     * Unmarks the task as incomplete.
+     */
+    public void unmarkTask() {
+        this.isDone = false;
+    }
+
+    /**
+     * Displays formatted information about the task.
+     * @return String containing common information about a task.
+     */
+    public String getStatusIcon() {
+        return this.isDone ? "[X] " + this.getDescription() : "[] " + this.getDescription();
+    }
+
+    /**
+     * Gets the description of the task.
+     * @return Description of the task in string.
+     */
+    public String getDescription() {
+        return this.description;
+    }
+
+    /**
+     * Encodes information about the task to be saved.
+     * @return String representation of the encoded information.
+     */
+    public String encode() {
+        return String.format("%s ### %s", this.isDone, this.description);
+    }
+
+    /**
+     * Checks if a task falls on a given date.
+     * @param date Date to check against.
+     * @return boolean value that checks if the task falls on the specified date.
+     */
+    public boolean fallsOnDate(LocalDate date) {
```

Comment:
Should the method name be changed to one that sounds more like a boolean? 👍

Reply:
I think this already suggests a boolean? The method tells us whether a task indeed falls on a given date. Trying to intentionally add the conventional "is" / "can" / "has" sounds a little too forced and grammatically-off. Unless you have a better name for it in mind? haha